lib/src/components/NxList/types.ts

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import { ForwardRefExoticComponent, RefAttributes, HTMLAttributes, LiHTMLAttributes, ValidationMap,
+  ReactNode, ButtonHTMLAttributes, AnchorHTMLAttributes, DetailedHTMLProps, FunctionComponent} from 'react';
+import PropTypes from 'prop-types';
+
+export interface NxListComponent
+  extends ForwardRefExoticComponent<NxListProps & RefAttributes<HTMLUListElement>> {
+  Item: ForwardRefExoticComponent<NxListItemProps & RefAttributes<HTMLLIElement>>;
+  Text: ForwardRefExoticComponent<NxListTextProps & RefAttributes<HTMLSpanElement>>;
+  Subtext: ForwardRefExoticComponent<NxListSubtextProps & RefAttributes<HTMLSpanElement>>;
+  Actions: ForwardRefExoticComponent<NxListActionProps & RefAttributes<HTMLDivElement>>;
+  ButtonItem: ForwardRefExoticComponent<NxListButtonItemProps & RefAttributes<HTMLLIElement>>;
+  LinkItem: ForwardRefExoticComponent<NxListLinkItemProps & RefAttributes<HTMLLIElement>>;
+  DescriptionTerm: FunctionComponent<DetailedHTMLProps<HTMLAttributes<HTMLElement>, HTMLElement>>;
+  Description: FunctionComponent<DetailedHTMLProps<HTMLAttributes<HTMLElement>, HTMLElement>>;
+}
+
+export interface NxListProps extends HTMLAttributes<HTMLUListElement> {
+  bulleted?: boolean | null;
+  children?: ReactNode;
+  emptyMessage?: ReactNode;
+  error?: string | null;
+  isLoading?: boolean | null;
+  retryHandler?: (() => void) | null;
+}
+
+export const nxListPropTypes: ValidationMap<NxListProps> = {
+  bulleted: PropTypes.bool,
+  children: PropTypes.node,
+  emptyMessage: PropTypes.node,
+  error: PropTypes.string,
+  isLoading: PropTypes.bool,
+  retryHandler: PropTypes.func
+};
+
+export interface NxListItemProps extends LiHTMLAttributes<HTMLLIElement> {
+  selected?: boolean | null;
```

Comment:
`NxList.Item` doesn't actually take this though right?

Reply:
> `NxList.Item` doesn't actually take this though right?

This entire `interface` and `propType` definition has been removed as `NxList.Item` is now defined using `withClass`